### PR TITLE
added some load functions for looking up RawlsUsers and Groups

### PR DESCRIPTION
**Pull request by helgridly**
Tiny PR that @MatthewBemis and I need to do other tasks.
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell  that the PR exists if he wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x]  sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [ ] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop 
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [x] **Submitter**: Verify swagger UI on dev server still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by helgridly** [comment]
@dvoet I added a function to turn an email address into an `Either[RawlsUser, RawlsGroup]`

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/AuthDAO.scala [hunk 1]
```diff
@@ -1,15 +1,34 @@
 package org.broadinstitute.dsde.rawls.dataaccess
 
+import org.broadinstitute.dsde.rawls.RawlsException
 import org.broadinstitute.dsde.rawls.model.WorkspaceAccessLevels.WorkspaceAccessLevel
 import org.broadinstitute.dsde.rawls.model._
 
+import scala.util.{Failure, Try}
+
 trait AuthDAO {
+  def loadUser(ref: RawlsUserRef, txn: RawlsTransaction): RawlsUser
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/AuthDAO.scala]: should these load functions return options?

Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/AuthDAO.scala]: `¯\_(ツ)_/¯`

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/AuthDAO.scala [hunk 2]
```diff
@@ -1,15 +1,34 @@
...
+
+  def loadUserByEmail(userEmail: String, txn: RawlsTransaction): RawlsUser
+
   def saveUser(rawlsUser: RawlsUser, txn: RawlsTransaction): RawlsUser
 
+  def loadGroup(groupRef: RawlsGroupRef, txn: RawlsTransaction): RawlsGroup
+
+  def loadGroupByEmail(groupEmail: String, txn: RawlsTransaction): RawlsGroup
```
Comment (MatthewBemis) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/AuthDAO.scala]: should all of these loads be Option[RawlsThing]?

Comment (MatthewBemis) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/AuthDAO.scala]: damnit doug beat me

**Comment by helgridly** [comment]
@dvoet @MatthewBemis options now instead

**Comment by MatthewBemis** [comment]
:+1:

**Comment by dvoet** [comment]
:+1:

**Pull request merged**